File: tobas_core/tobas_rqt_py/tobas_rqt_py/widgets/table_widget.py
from PyQt5.QtWidgets import QTableWidget


class TableWidget(QTableWidget):
    """
    ===== QTableWidgetとの違い =====
    - 追加メソッド
    """

    def remove_all(self) -> None:
        """全ての行を削除する．clearとは異なり，内容に加えセルまで削除する．"""
        while self.rowCount() > 0:
            self.removeRow(0)

File: tobas_gui/tobas_qt_tools/src/widgets/table_widget.cpp
// SPDX-License-Identifier: GPL-3.0-or-later
// Copyright (C) 2026 Tobas, Inc.

#include "tobas_qt_tools/widgets/table_widget.hpp"

#include <QHeaderView>

namespace tobas
{
namespace qt
{
void TableWidget::removeAll()
{
  while (rowCount() > 0) {
    removeRow(0);
  }
}

void TableWidget::setColumnsWidth(int width)
{
  for (int col = 0; col < columnCount(); ++col) {
    setColumnWidth(col, width);
  }
}

void TableWidget::setHeaderSectionsClickable(bool clickable)
{
  horizontalHeader()->setSectionsClickable(clickable);
  verticalHeader()->setSectionsClickable(clickable);
}
}  // namespace qt
}  // namespace tobas
